pg_search/src/api/operator/text.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2023-2024 Retake, Inc.
+//
+// This file is part of ParadeDB - Postgres for Search and Analytics
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use crate::api::operator::{
+    anyelement_text_opoid, estimate_selectivity, make_search_config_opexpr_node,
+    ReturnedNodePointer,
+};
+use crate::postgres::utils::locate_bm25_index;
+use crate::query::SearchQueryInput;
+use crate::schema::SearchConfig;
+use crate::UNKNOWN_SELECTIVITY;
+use pgrx::pg_sys::planner_rt_fetch;
+use pgrx::{is_a, pg_extern, pg_sys, AnyElement, FromDatum, Internal, PgList};
+
+#[pg_extern(immutable, parallel_safe)]
+pub fn search_with_text(
+    _element: AnyElement,
+    query: &str,
+    _fcinfo: pg_sys::FunctionCallInfo,
+) -> bool {
+    panic!("query is incompatible with pg_search's `@@@(key_field, TEXT)` operator: `{query}`")
+}
```

Comment:
It's the function behind the `@@@(anyelement, text)` operator.  Since we transform those to use `@@@(anyelement, jsonb)`, this function won't be called in normal circumstances, but it could be called if the rhs of the `@@@` is some kind of volatile value.  And in that case we just have to give up.

Reply:
I added some comments here.